Xml config test failures

maintenance-3.x (commit a82a7feeda0ecb5a466442e21aea9d2f76aeb48f)

http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-3.maintenance-OracleJDK6/25/consoleFull

```
10:29:14 [ERROR] testResolveSystemProperty_file_xml(com.hazelcast.config.XmlConfigBuilderConfigResolutionTest)  Time elapsed: 0.051 s  <<< ERROR!
10:29:14 java.lang.NoClassDefFoundError: java/nio/file/Paths
10:29:14 	at com.hazelcast.config.helpers.DeclarativeConfigFileHelper.ensureTestConfigDeleted(DeclarativeConfigFileHelper.java:230)
10:29:14 	at com.hazelcast.config.XmlConfigBuilderConfigResolutionTest.beforeAndAfter(XmlConfigBuilderConfigResolutionTest.java:50)
10:29:14 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
10:29:14 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
10:29:14 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
10:29:14 	at java.lang.reflect.Method.invoke(Method.java:597)
10:29:14 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
10:29:14 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
10:29:14 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
10:29:14 	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:402)
10:29:14 	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)
10:29:14 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
10:29:14 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
10:29:14 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
10:29:14 	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
10:29:14 	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
10:29:14 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
10:29:14 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
10:29:14 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
10:29:14 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
10:29:14 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
10:29:14 	at com.hazelcast.test.AbstractHazelcastClassRunner$1.evaluate(AbstractHazelcastClassRunner.java:302)
10:29:14 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
10:29:14 	at org.junit.runners.Suite.runChild(Suite.java:128)
10:29:14 	at org.junit.runners.Suite.runChild(Suite.java:27)
10:29:14 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
10:29:14 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
10:29:14 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
10:29:14 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
10:29:14 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
10:29:14 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
10:29:14 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
10:29:14 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
10:29:14 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
10:29:14 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
10:29:14 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
10:29:14 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
10:29:14 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:379)
10:29:14 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:340)
10:29:14 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:125)
10:29:14 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:413)
10:29:14 Caused by: java.lang.ClassNotFoundException: java.nio.file.Paths
10:29:14 	at java.net.URLClassLoader$1.run(URLClassLoader.java:202)
10:29:14 	at java.security.AccessController.doPrivileged(Native Method)
10:29:14 	at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
10:29:14 	at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
10:29:14 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
10:29:14 	at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
10:29:14 	... 41 more
```

--- a/hazelcast/src/test/java/com/hazelcast/config/helpers/DeclarativeConfigFileHelper.java
+++ b/hazelcast/src/test/java/com/hazelcast/config/helpers/DeclarativeConfigFileHelper.java
@@ -19,6 +19,8 @@ package com.hazelcast.config.helpers;
 import com.hazelcast.config.Config;
 
 import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
 import java.io.PrintWriter;
 import java.net.URL;
 import java.util.LinkedList;
@@ -226,11 +228,26 @@ public class DeclarativeConfigFileHelper {
                 + "    - hazelcast-client-c1.yaml";
     }
 
-    public void ensureTestConfigDeleted() {
+    public void ensureTestConfigDeleted() throws IOException {
         if (!testConfigPaths.isEmpty()) {
             for (String testConfigPath : testConfigPaths) {
-                File file = new File(testConfigPath);
-                file.delete();
+                if (testConfigPath != null) {
+                    File file = new File(testConfigPath);
+
+                    if (!file.exists()) {
+                        throw new FileNotFoundException(testConfigPath + " couldn't be deleted because it doesn't exist");
+                    }
+
+                    if (!file.canWrite()) {
+                        throw new IOException(testConfigPath + " couldn't be deleted because write is not permitted");
+                    }
+
+                    if (!file.delete()) {
+                        throw new IOException(
+                                testConfigPath + " couldn't be deleted. On Windows it most likely means the files is "
+                                        + "still used.");
+                    }
+                }
             }
         }
     }

